picard/metadata.py

```diff
@@ -244,6 +247,35 @@ def compare_to_release_parts(self, release, weights):
         except (ValueError, KeyError):
             pass
 
+        # Date Logic - in order of preference
```

Comment:
It would be nice to move these comments near the actual else cases as it allows us to understand what's happening.

Reply:
done